File: todolist/src/main/java/com/oreooo/todolist/LItePalDB/LitePalHelper.java
package com.oreooo.todolist.LItePalDB;

import com.example.oreooo.todoforstudy.LItePalDB.Project;

import org.litepal.LitePal;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class LitePalHelper {

    public static LitePalHelper getInstance() {
        return LitepalHelperHolder.Instance;
    }

    private static class LitepalHelperHolder {
        private static LitePalHelper Instance = new LitePalHelper();
    }

    public void addProject (Project project) {
        project.save();
    }

    public void updateProject (Project project) {
        project.updateAll("addTime=? and thePlan=?", project.getAddTime(), project.getThePlan());
    }

    public List<Project> getAllProject () {
        List data = new ArrayList<>();
        data = Arrays.asList(LitePal.findAll(Project.class).toArray());
        return data;
    }

    public List<Project> getNotDoneProject () {
        List data = new ArrayList<>();
        data = LitePal.where("done = ?", "false")
                .order("doneTime DESC")
                .find(Project.class);
        return data;
    }

    public List<Project> getDoneProjectsByToday (String doneDate) {
        List data = new ArrayList<>();
        data = LitePal.where("doneDate = ?", doneDate)
                .order("doneTime")
                .find(Project.class);
        return data;
    }
}

File: module_todolist/src/main/java/com/oreooo/todolist/lItepal/LitePalHelper.kt
package com.oreooo.todolist.lItepal

import org.litepal.LitePal
import org.litepal.extension.findAll
import java.util.*

class LitePalHelper {

    val allProject: List<Project>
        get() {
            var data: List<Project>
//            data = Arrays.asList<Project>(*LitePal.findAll(Project::class.java).toTypedArray())
            data = LitePal.findAll<Project>()
            return data
        }

    val notDoneProject: List<Project>
        get() {
            var data: List<Project>
            data = LitePal.where("done=?", "false")
                    .order("doneTime DESC")
                    .find(Project::class.java)
            return data
        }

    private object LitepalHelperHolder {
        private val Instance = LitePalHelper()
    }

    fun addProject(project: Project) {
        project.save()
    }

    fun updateProject(project: Project) {
        project.updateAll("addTime=? and thePlan=?", project.addTime, project.thePlan)
    }

    fun getDoneProjectsByToday(doneDate: String): List<Project> {
        var data: List<*>
        data = LitePal.where("doneDate=?", doneDate)
                .order("doneTime")
                .find(Project::class.java)
        return data
    }

    companion object {
        val instance: LitePalHelper by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED) {
            LitePalHelper()
        }
    }
}